### PR TITLE
Feature/add health endpoint

### LINKED ISSUE #156: Add health endpoint
#### Description

Add health endpoint that details the current status of the metadata submitter. the status can be dependent on existing database connection, load times etc.
health endpoint is not restricted to logged in user

#### DoD (Definition of Done)

`/health` endpoint has various status of Ok, Partially down (in case db is down or no connection to dependent services), or Degraded if load times are slow

#### Testing

Unit testing, integration testing and pee review

### PR DESCRIPTION
**Pull request by csc-jm**
### Description

PR adds a health check endpoint.

### Related issues

Fixes #156 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Added a health check endpoint
- Added tests for the endpoint

### Testing

- [x] Unit Tests
- [x] Integration Tests

### Mentions

Health check format is based on the one found here in [swift-browser-ui](https://github.com/CSCfi/swift-browser-ui).

### CODE REVIEW

File: tests/integration/run_tests.py [hunk 1]
```diff
@@ -596,15 +606,6 @@ async def main():
...
 
-        # Test add, modify, validate and release action with submissions
-        LOG.debug("=== Testing actions within submissions ===")
-        submission_folder = {
-            "name": "submission test 1",
-            "description": "submission test folder 1",
-        }
-        submission_folder_id = await post_folder(sess, submission_folder)
-        await test_submissions_work(sess, submission_folder_id)
-
```
Comment (csc-jm) [inline tests/integration/run_tests.py:607]: This seemed like a duplicate test so removed it... unless there's a reason for it to be there?

Comment (blankdots) [reply tests/integration/run_tests.py:607]: that was there for a purpose to show that objects separation is done for each user

Comment (csc-jm) [reply tests/integration/run_tests.py]: reverted this commit so it's back in there

File: metadata_backend/api/health.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class HealthHandler:
+    """Handler for health check."""
+
+    async def get_health_status(self, req: Request) -> Response:
+        """Check health status of the application and return a JSON object portraying the status.
+
+        :param req: GET request
+        :returns: JSON response containing health statuses
+        """
+        db_client = req.app["db_client"]
```
Comment (blankdots) [inline metadata_backend/api/health.py:22]: to avoid the 15 retry delay can we create a new AsyncIOMotorClient that uses the `url` from config ?

Comment (csc-jm) [reply metadata_backend/api/health.py]: Attempted this in the latest commits

**Review by blankdots — APPROVED** [review]

**Pull request merged**